### PR TITLE
[CSL-3168] Surface the Display Name of Recommendations Pod by Default

### PR DESCRIPTION
**Pull request by Mudaafi**
## Summary
Recommendations Pod have a "display_name" attribute that we would want to render by default if available. If the user has specified a `displayName` configuration for the pod via the library, we should use that instead.
The css classnames should still reflect pod-id to avoid weird looking classnames such as `Best Sellers` rather than `best-sellers`.

## Changes Made
- `useFetchRecommendationsPod` now returns an additional object `podsData` which contains both the `podId` and `displayName` returned by the response.
- Consequently, `utils/getActiveSectionsWithData` now takes in the `podsData` as a new optional parameter.
- In `getSectionProps` in `useCioAutocomplete`, `sectionTitle` is used setting classnames and should reflect `podId` rather than `displayName`
- In `SectionItemList`, `sectionTitle` here is used to render to the end user and should reflect the `displayName` for Recs Pods, if available.

### CODE REVIEW

**Review by esezen — COMMENTED** [review]
This is looking good. The only thing I'm wondering is if there is a way to make `podsData` part of the other objects instead of passing it around individually. I couldn't find a good solution though, so I think what we have here is good. Just wanted to mention it to see if there are any other ideas. Not marking it as approved to make sure we get a review from @mocca102 before we merge but this LGTM!

File: src/hooks/useCioAutocomplete.ts
Comment (mocca102) [inline src/hooks/useCioAutocomplete.ts:216]: Can we rename this to sectionClassName instead? or anything that says this differs from the section's actual title? The name is confusing and coming back to this in the future we would forget that this is only for the class name

File: src/hooks/useFetchRecommendationPod.ts
Comment (mocca102) [inline src/hooks/useFetchRecommendationPod.ts:33]: We already did it once and added the podId. Can we also add sectionDisplayName here as well instead of having separate podData?

File: src/hooks/useCioAutocomplete.ts
Comment (Mudaafi) [reply src/hooks/useCioAutocomplete.ts:216]: I'm in total agreement. @esezen I think you worked on this previously, but would `sectionClassName` work for you as well?

File: src/utils.ts
Comment (mocca102) [inline src/utils.ts:157]: This is where we have the sections configs and I believe it's the best place we can add the displayName for the recommendations active sections
Screenshot 2024-01-24 at 11 27 15 PM

File: src/hooks/useSections.ts
Comment (mocca102) [inline src/hooks/useSections.ts:62]: Here is the place where we can get the data from recommendationsResults where the display name on the items and add it the appropriate active section instead of returning podsData then passing it

**Review by mocca102 — CHANGES_REQUESTED** [review]
I left some comments. Let me know what do you think. It's just about how we pass the data around

File: src/hooks/useFetchRecommendationPod.ts
Comment (Mudaafi) [reply src/hooks/useFetchRecommendationPod.ts:33]: We could but I don't particularly like it mainly because the sectionDisplayName would exist per item rather than at the top level which could get messy. From what I understand, `podId` needs to exist per item for tracking-related things but `displayName` doesn't so I would rather not put it there.

An alternative implementation, and this might attend to Enes' idea, we could modify the `recommendationsPodResults` to be an object rather than an array. That way we can return one object rather than two but this will be change the expected interface.

Comment (mocca102) [reply src/hooks/useFetchRecommendationPod.ts:33]: I agree that modifying `recommendationsPodResults` would be the ideal thing here so I thought adding sectionDisplayName would be less intrusive and wouldn't require passing extra parameters.

I'm ok if you want to keep it the way you implemented though

File: src/hooks/useCioAutocomplete.ts
Comment (Mudaafi) [reply src/hooks/useCioAutocomplete.ts:216]: We'll track this on PR #128 instead

File: src/hooks/useFetchRecommendationPod.ts
Comment (Mudaafi) [reply src/hooks/useFetchRecommendationPod.ts:33]: We'll opt of merging it before calling `setActiveSectionsWithData`

**Review by mocca102 — APPROVED** [review]
Thanks for making the changes. LGTM!

File: src/utils.ts
Comment (esezen) [inline src/utils.ts:185]: @Mudaafi let me know if I'm thinking about this incorrectly but I think order of the spread (...) vs `data` is important here. For custom sections, we want new `sectionData` to override the one coming from the config.
Screenshot 2024-01-27 at 1 34 45 PM
Screenshot 2024-01-27 at 1 33 49 PM

**Review by esezen — CHANGES_REQUESTED** [review]

File: src/utils.ts
Comment (mocca102) [reply src/utils.ts:185]: Should sectionConfig have data for other sections than custom? I think it shouldn't 🤔

Comment (esezen) [reply src/utils.ts:185]: Correct, only for `custom`

Comment (mocca102) [reply src/utils.ts:185]: So why would we worry if sectionConfig overrides sectionData then?

Comment (esezen) [reply src/utils.ts:185]: Because we don't want that for custom sections and this code runs for custom sections? The screenshot above is from a custom section. Am I understanding this wrong?

Comment (mocca102) [reply src/utils.ts:185]: What I mean is it's ok for custom sections if it overrides the data since we're not handling it anyway

Comment (esezen) [reply src/utils.ts:185]: On line 173 we are modifying the data being passed to us then here we are overriding it back

Comment (mocca102) [reply src/utils.ts:185]: Aha got it

Comment (Mudaafi) [reply src/utils.ts:185]: Just so I'm on the same page, are we saying the order of priority is:
1. Dashboard/Catalog set configs
2. Developer-set top-level-component configs

That didn't feel intuitive to me :thinking: but maybe it's because I'm thinking of it as an "override" rather than a "fallback"

Comment (Mudaafi) [reply src/utils.ts:185]: Okay nvm scratch that, I get the issue with `data` getting overriden

**Review by esezen — APPROVED** [review]
LGTM!

**Pull request merged**
